mu/app.py

```diff
@@ -59,16 +64,25 @@
 
 class AnimatedSplash(QSplashScreen):
     """
-    An animated splash screen for gifs.
+    An animated splash screen for gifs. Includes a text area for logging
+    output.
     """
 
     def __init__(self, animation, parent=None):
         """
         Ensure signals are connected and start the animation.
         """
-        super().__init__()
+        self.log_lines = 4
+        # To hold only number of log_lines of logs to display.
+        self.log = []
```

Comment:
Tiny nitpick, but I assume this object never gets destructed, the widget is "just closed", so we could clean up things like these that would otherwise stay in memory.

Reply:
OK, so in Qt there's a `deleteLater` call that can be made to "garbage collect" (not strictly true, but you get what I mean) a QObject. I think, if I call that, everything to do with this object will be gone from memory. In fact, if I enclose the splash work in its own function and call `deleteLater` in the function, then the Python references will be garbage collected when we leave the function scope and Qt will destroy the SplashScreen object.